Sources/Fluent/Relations/Siblings.swift

```diff
@@ -1,47 +1,83 @@
-public final class Siblings<T: Entity> {
-    fileprivate let _query: Query<T>
-    fileprivate let _left: Entity.Type
+/// Represents a many to many relationship
+/// through a Pivot table from the Local 
+/// entity to the Foreign entity.
+public final class Siblings<
+    Local: Entity, Foreign: Entity
+> {
+    /// The id of the Local entity.
+    ///
+    /// This will be used to filter the 
+    /// collection of foreign entities related
+    /// to the local entity type.
+    let id: Node
 
-    public let localKey: String
-    public let foreignKey: String
+    /// Foreign entity key that is foreign
+    /// to the entity it resides on
+    ///
+    /// This id is a pointer to "Local" keys
+    ///
+    /// ex: "foo_id"
+    let foreignKey: String
 
-    public init<E: Entity>(entity: E, localKey: String?, foreignKey: String?) throws {
+    /// Foreign entity key that is local
+    /// to the entity it resides on
+    ///
+    /// "Foreign" keys point to this id.
+    ///
+    /// ex: "id"
+    let localKey: String
+
+    /// Create a new Siblings relationsip using 
+    /// a Local and Foreign entity.
+    ///
+    /// See Siblings.localKey and Siblings.foreignKey
+    /// for more information about how to use them.
+    public init(
+        _ entity: Local,
+        foreignKey: String = Foreign.foreignIdKey,
+        localKey: String = Foreign.idKey
+    ) throws {
         guard let ident = entity.id else {
             throw RelationError.noIdentifier
         }
 
-        let query = try T.query()
-
-        let localKey = localKey ?? T.idKey 
-        let foreignKey = foreignKey ?? "\(T.name)_\(T.idKey)"
-
-        self.localKey = localKey
+        id = ident
```

Comment:
I think just id and `self.id = id` to have parity w/ other variables initialized